test/inductor/test_compiled_optimizers.py

```diff
@@ -194,9 +231,14 @@ def make_test(
     def test_fn(self):
         stack = ExitStack()
         try:
+            # we fallback to eager on the fused implementation
+            is_fused = kwargs.get("fused", False)
```

Comment:
What's the difference between the fused and the unsupported forloop impl? To my understanding, they both fall back to eager--so could we reuse the kernel_count value here instead of specializing on fused?

Reply:
So since singletensor has its own entry in the kernel count tuple, I just change the kernel count there to indicate unsupported. Since there's only two fused optimizers, it doesn't have a dedicated entry in the kernel count tuples. That's the reason for this carveout. I think it's good to check the kernel count is zero though since we don't check that above.

Reply:
These should logically live together--the fewer carveouts the more readable/coherent!

Reply:
Yeah I just removed this